File: JAVA/large_fibonacci_numbers.java
import java.math.BigInteger;
import java.util.*;

public class large_fibonacci_numbers {
	public static void main(String[] args) {
		BigInteger a= new BigInteger("1");
		BigInteger b= new BigInteger("1");
		BigInteger c=new BigInteger("0");
		Scanner s= new Scanner(System.in);
		System.out.println("Enter a number:");
        int n= s.nextInt();

		long startTime = System.nanoTime();
		for(int i=0;i<n-2;i++) {
			c=a.add(b);
			a=b;
			b=c;
		}
		long endTime = System.nanoTime();
		long duration = endTime - startTime;
		System.out.println("Execution time: " +(duration/1000000) + " milliseconds");
		//System.out.println(c);
		System.out.println(c.toString().length());
		s.close();
	}
}

File: C++/large_fibonacci_numbers.cpp
#include<iostream>
#include <chrono>
#include <bits/stdc++.h>
using namespace std;
#define ull unsigned long long

int main(){
    ull a=1,b=1,c;
    int n;
    cout<<"Enter n for nth fibonacci number:"<<endl;
    cin>>n;
    cout<<a<<endl;
    cout<<b<<endl;
    auto startTime = chrono::high_resolution_clock::now();
    for(int i=0;i<n-2;i++){
        c=a+b;
        a=b;
        b=c;
        cout<<c<<endl;
    }
    auto endTime = chrono::high_resolution_clock::now();
    auto duration = chrono::duration_cast<chrono::milliseconds>(endTime - startTime).count();
    cout << "Execution time: " << duration << " milliseconds" << endl;
    cout << to_string(c).length() << endl;
    return 0;
}